### DISCUSSION TITLE
useEffect cleanup not called when navigating to `.mdx` Story

### DISCUSSION THREAD

**Discussion by DannyJoris** [storybookjs/storybook#18627] [Help]
I have a React Modal component which makes use of React Portal.

When I navigate from the Modal Story to any `.mdx` Story, the modal doesn't get removed. I tried setting the state to `false` on `useEffect` cleanup. The cleanup function does not get called. It does get called when navigating to other Stories. That said, the component never re-renders. But setting state via `useEffect` isn't necessary when navigating to other Stories anyways.
```
const Template: ComponentStory<typeof Modal> = (args) => {
  const [showDialog, setShowDialog] = useState(true);
  useEffect(() => {
    return () => {
      console.log('cleanup');
      setShowDialog(false);
    };
  }, [setShowDialog]);
  return <Modal {...args} showDialog={showDialog} close={setShowDialog} />;
};
```

Thoughts?

**Comment by mauris** [comment]
observing the same, that component's `useEffect` clean up function doesn't get called when navigating from the component's story to the `Docs` tab or to another MDX Story.
